apps/hash-graph/docker/Dockerfile

```
@@ -1,20 +1,42 @@
-FROM alpine:3.17 as rust
+FROM node:16.18.1-alpine AS rust
+
+WORKDIR /usr/local/
 
 ENV RUSTUP_HOME=/usr/local/rustup \
     CARGO_HOME=/usr/local/cargo \
     PATH=:$PATH:/usr/local/cargo/bin
 
-COPY rust-toolchain.toml .
+COPY apps/hash-graph/rust-toolchain.toml .
 RUN apk add --no-cache gcc musl-dev bash && \
     wget -q -O- https://sh.rustup.rs | sh -s -- -y --default-toolchain none --profile minimal && \
     rustup show
 
 SHELL ["bash", "-c"]
 
+FROM rust as node_base
+
+WORKDIR /usr/local/
+
+# Ensure that the node module layer can be cached
+COPY package.json .
+COPY yarn.lock .
+
+COPY ./libs/@local/status/. /usr/local/libs/@local/status
+
+COPY libs/@local/eslint-config libs/@local/eslint-config
+COPY libs/@local/tsconfig libs/@local/tsconfig
+
+RUN yarn workspace @local/status install --frozen-lockfile --ignore-scripts --prefer-offline
+
+FROM node_base AS builder
```

Having two blank lines makes it easier to find new layers
```suggestion

FROM node_base AS builder
```

---

```suggestion

FROM node_base AS builder
```